mimic/test/test_auth.py

```diff
@@ -625,6 +625,40 @@ def test_token_and_catalog_for_password_credentials_wrong_tenant(self):
             }
         })
 
+    def test_rax_kskey_apikeycredentials(self):
+        """
+        Test apiKeyCredentials
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+        root = MimicRoot(core).app.resource()
+        (response, json_body) = self.successResultOf(json_request(
+            self, root, "GET",
+            "/identity/v2.0/users/1/OS-KSADM/credentials/RAX-KSKEY:apiKeyCredentials"
+        ))
+        self.assertEqual(response.code, 404)
+        self.assertEqual(json_body['itemNotFound']['message'], 'User 1 not found')
+        creds = {
+            "auth": {
+                "passwordCredentials": {
+                    "username": "HedKandi",
+                    "password": "Ministry Of Sound UK"
+                },
+                "tenantId": "77777"
+            }
+        }
+        (response, json_body) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens", creds))
+        self.assertEqual(response.code, 200)
+        user_id = json_body['access']['user']['id']
+        username = json_body['access']['user']['name']
+        (response, json_body) = self.successResultOf(json_request(
+            self, root, "GET",
+            "/identity/v2.0/users/" + user_id + "/OS-KSADM/credentials/RAX-KSKEY:apiKeyCredentials"
+        ))
+        self.assertEqual(response.code, 200)
+        self.assertEqual(json_body['RAX-KSKEY:apiKeyCredentials']['username'],
+                         username)
```

Comment:
This test should probably assert something about the API key returned in the response, although it's probably inappropriate to check for an exact match against the source code. Maybe use a regex to assert that it's a 32 chars hex string?